src/main/scala/play/api/libs/mailer/MailerPlugin.scala

```diff
@@ -109,6 +111,14 @@ class SMTPMailer @Inject() (smtpConfiguration: SMTPConfiguration) extends Mailer
   }
 }
 
+class SMTPReconfigurableMailer @Inject()(smtpConfigurationProvider: Provider[SMTPConfiguration]) extends MailerClient {
```

Comment:
I think `SMTPDynamicMailer` is a better naming (as opposed to a static mailer client where the configuration is static).